sql-plugin/src/main/scala/org/apache/spark/sql/rapids/collectionOperations.scala

```diff
@@ -392,4 +392,236 @@ case class GpuArrayMax(child: Expression) extends GpuBaseArrayAgg with ImplicitC
+      stop: GpuColumnVector,
+      step: GpuColumnVector): ColumnVector = {
+    withResource(calculateSize(start.getBase, stop.getBase, step.getBase, start.getRowCount.toInt))
+    { size => ColumnVector.sequence(start.getBase, size, step.getBase) }
+  }
+
+  override def doColumnar(
+      start: GpuScalar,
+      stop: GpuColumnVector,
+      step: GpuColumnVector): ColumnVector = {
+    withResource(calculateSize(start.getBase, stop.getBase, step.getBase, stop.getRowCount.toInt)) {
+      size =>
+      withResource(ColumnVector.fromScalar(start.getBase, stop.getRowCount.toInt)) { startV =>
+        ColumnVector.sequence(startV, size, step.getBase)
+      }
+    }
+  }
+
+  override def doColumnar(
+      start: GpuScalar,
+      stop: GpuScalar,
+      step: GpuColumnVector): ColumnVector = {
+    withResource(ColumnVector.fromScalar(start.getBase, step.getRowCount.toInt)) { startV =>
+      withResource(calculateSize(startV, stop.getBase, step.getBase, step.getRowCount.toInt)) {
+        size => ColumnVector.sequence(startV, size, step.getBase)
+      }
+    }
+  }
+
+  override def doColumnar(
+      start: GpuScalar,
+      stop: GpuColumnVector,
+      step: GpuScalar): ColumnVector = {
+    withResource(calculateSize(start.getBase, stop.getBase, step.getBase, stop.getRowCount.toInt)) {
+      size =>
+        withResource(ColumnVector.fromScalar(start.getBase, stop.getRowCount.toInt)) { startV =>
+          withResource(ColumnVector.fromScalar(step.getBase, stop.getRowCount.toInt)) { stepV =>
+            ColumnVector.sequence(startV, size, stepV)
+          }
+        }
+    }
+  }
+
+  override def doColumnar(
+      start: GpuColumnVector,
+      stop: GpuScalar,
+      step: GpuColumnVector): ColumnVector = {
+    withResource(calculateSize(start.getBase, stop.getBase, step.getBase, start.getRowCount.toInt))
+    { size => ColumnVector.sequence(start.getBase, size, step.getBase) }
+  }
+
+  override def doColumnar(
+      start: GpuColumnVector,
+      stop: GpuScalar,
+      step: GpuScalar): ColumnVector = {
+    withResource(calculateSize(start.getBase, stop.getBase, step.getBase, start.getRowCount.toInt))
+    { size =>
+      withResource(ColumnVector.fromScalar(step.getBase, start.getRowCount.toInt)) { stepV =>
+        ColumnVector.sequence(start.getBase, size, stepV)
+      }
+    }
+  }
+
+  override def doColumnar(
+      start: GpuColumnVector,
+      stop: GpuColumnVector,
+      step: GpuScalar): ColumnVector =
+    withResource(calculateSize(start.getBase, stop.getBase, step.getBase, start.getRowCount.toInt))
+    { size =>
+      withResource(ColumnVector.fromScalar(step.getBase, start.getRowCount.toInt)) { stepV =>
+        ColumnVector.sequence(start.getBase, size, stepV)
+      }
+    }
+
+  override def doColumnar(
+      numRows: Int,
+      val0: GpuScalar,
+      val1: GpuScalar,
+      val2: GpuScalar): ColumnVector =
+    throw new IllegalStateException("This is not supported yet")
```

Comment:
Hmm, Could you add the support when we come across this kind of odd corner cases? since I don't know how to simulate this corner case.

Reply:
Just turn `val0`/`start` into a `GpuColumnVector` and then call the other `doColumnar`.  That way you don't really need to test it that much.  The corner cases show up when a constant is injected into a plan after constant folding has finished, this happens some times but it is hard to reproduce.  It can also happen when some operations, like `GpuCoalesce` where Spark cannot determine that a constant value is always returned by the coalesce, but we can.

For example...
```
sequence(coalesce(100, some_column), 200, 10)
```

I think will cause this to happen. This is the same thing we need for the version of sequence without a step.

Reply:
Thx. Done